src/layers/basic.jl

```diff
@@ -226,3 +226,85 @@ end
 function Base.show(io::IO, b::SkipConnection)
   print(io, "SkipConnection(", b.layers, ", ", b.connection, ")")
 end
+
+"""
+    Bilinear(in1, in2, out)
+
+Creates a Bilinear layer, that operates over two inputs at the same time.
+Given parameters W and b, it's output is an interaction of the form 
+
+z_i = σ.(x * W_i * y' .+ b_i)
+
+All `z_i` are then concatenated across output dimensions `i` and samples.
+
+Bilinear can be used to create interactions between features as well.
+
+```
+d = Dense(10, 10)
+b = Bilinear(10, 10, 5)
+x = randn(Float32,10,9)
+
+#using Bilinear as the recombinator in a SkipConnection
+sc = SkipConnection(d, b)
+size(sc(x)) == (5,9)
+
+#using Bilinear in two data streams at once
+x = randn(Float32,10,9)
+y = randn(Float32,2,9)
+b = Bilinear(10, 2, 3)
+size(b(x,y)) == (3,9)
+
+#using Bilinear to generate interactions
+x = randn(Float32,11,7)
+b = Bilinear(11, 11, 3)
+size(b(x)) == (3,7)
+```
+"""
+struct Bilinear{A,B,S}
+  W::A
+  b::B
+  σ::S
+end
+
+@functor Bilinear
+
+Bilinear(W, b) = Bilinear(W, b, identity)
+
+function Bilinear(in1::Integer, in2::Integer, out::Integer, σ = identity;
+  initW = glorot_uniform, initb = zeros)
+  return Bilinear(initW(out, in1, in2), initb(out), σ)
+end
+
+function (a::Bilinear)(x::AbstractMatrix, y::AbstractMatrix)
+  if size(x, 2) ≠ size(y, 2)
+    error("Bilinear expected equal number of samples in both streams. Got $(size(x,2)) and $(size(y,2))")
+  end
```

Comment:
```suggestion
  end
  W, b, σ = a.W, a.b, a.σ
```
This got lost in copy-paste I think.